### PR TITLE
Al-2382 support AWP centralized scan mode

### PR DESCRIPTION
**Pull request by shayde78**
CloudGuard AWP (AWS) - Terraform Module

This pull request introduces the implementation of AWP Centralized scan mode for AWS

AWP (Agentless Workload Posture) for AWS accounts already enables SAAS and inAccount scan modes,
This pull request adds **inAccountHub** - for the centralized account, where the scans execute,
and **inAccountSub** for each account that should be scanned in the centralized account

for more info please visit
(https://www.checkpoint.com/dome9/)

This module uses [Check Point CloudGuard Dome9 Provider](https://registry.terraform.io/providers/dome9/dome9/latest/docs)

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -1,25 +1,36 @@
 variable "awp_cloud_account_id" {
-    description = "CLOUDGUARD_ACCOUNT_ID or EXTERNAL_AWS_ACCOUNT_NUMBER"
-    type        = string
+  description = "CLOUDGUARD_ACCOUNT_ID or EXTERNAL_AWS_ACCOUNT_NUMBER"
+  type        = string
+}
+
+variable "awp_hub_external_account_id" {
```
Comment (batyakopelowitz) [inline variables.tf:6]: rename it to awp_centralized_cloud_account_id
```suggestion
variable "awp_centralized_cloud_account_id" {
```

File: variables.tf [hunk 2]
```diff
@@ -1,25 +1,36 @@
...
+  description = "centralized account EXTERNAL_AWS_ACCOUNT_NUMBER"
+  type        = string
+  default     = null
+}
+
+variable "awp_organization_id" {
```
Comment (batyakopelowitz) [inline variables.tf:12]: Why do we need the client to give us the org id?
We can take it from Mongo

File: variables.tf [hunk 3]
```diff
@@ -1,25 +1,36 @@
 variable "awp_cloud_account_id" {
-    description = "CLOUDGUARD_ACCOUNT_ID or EXTERNAL_AWS_ACCOUNT_NUMBER"
-    type        = string
+  description = "CLOUDGUARD_ACCOUNT_ID or EXTERNAL_AWS_ACCOUNT_NUMBER"
+  type        = string
+}
+
+variable "awp_hub_external_account_id" {
+  description = "centralized account EXTERNAL_AWS_ACCOUNT_NUMBER"
```
Comment (batyakopelowitz) [inline variables.tf:7]: support also CLOUDGUARD_ACCOUNT_ID

File: versions.tf
[the remark this answers is not in the document]
Comment (shayde78) [reply versions.tf:1]: it is only "terraform fmt"...

File: README.md [hunk 4]
```diff
@@ -29,6 +29,7 @@ module "terraform-dome9-awp-aws" {
   # e.g:
```
Comment (yuval-avram) [inline README.md:29]: # The scan mode for the AWP. Valid values are "inAccount" or "saas".
  
  Add centralized and sub values

File: README.md [hunk 5]
```diff
@@ -29,6 +29,7 @@ module "terraform-dome9-awp-aws" {
...
   awp_cross_account_role_name        = "<CrossAccountRoleName>"
   awp_cross_account_role_external_id = "<ExternalId>"
+  awp_centralized_cloud_account_id   = "in case of centralized onboarding, set the 'AWS account id' of the central account where the scans executed"
   awp_additional_tags                = {}  # e.g {"key1" = "value1", "key2" = "value2"}
```
Comment (yuval-avram) [inline README.md:35]: If settings are not allowed in centralized/sub, please write it (and hopefully you validated it at the api)

File: README.md [hunk 6]
```diff
@@ -69,7 +70,9 @@ module "terraform-dome9-awp-aws" {
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
 | <a name="input_awp_cloud_account_id"></a> [awp_cloud_account_id](#input\_awp\_cloud\_account\_id) | The Id of the AWS account, onboarded to CloudGuard (can be either the Dome9 Cloud Account ID or the AWS Account Number) | `string` | n/a | yes |
-| <a name="input_awp_scan_mode"></a> [awp_scan_mode](#input\_awp\_scan\_mode) | The scan mode for the AWP `[ "inAccount" \| "saas" ]`| `string` | "inAccount" | yes |
+| <a name="input_awp_centralized_cloud_account_id"></a> [awp_centralized_cloud_account_id](TODO) | The Id of the central AWS account where the scans take place  | `string` | n/a | in case of inAccountSub scan mode |
+| <a name="awp_organization_id"></a> [awp_organization_id](TODO) | The AWS organization id in case of centralized account  | `string` | n/a | in case of inAccountHub scan mode |
```
Comment (yuval-avram) [inline README.md:74]: As discussed - please remove

File: README.md [hunk 7]
```diff
@@ -69,7 +70,9 @@ module "terraform-dome9-awp-aws" {
...
+| <a name="input_awp_scan_mode"></a> [awp_scan_mode](#input\_awp\_scan\_mode) | The scan mode for the AWP `[ "inAccount" \| "saas" \| "inAccountHub" \| "inAccountSub" ]`| `string` | "inAccount" | yes |
 | <a name="input_awp_cross_account_role_name"></a> [awp_cross_account_role_name](#input\_awp\_cross\_account\_role\_name) | AWP Cross account role name | `string` | `CloudGuardAWPCrossAccountRole` | no |
 | <a name="input_awp_cross_account_role_external_id"></a> [awp_cross_account_role_external_id](#input\_awp\_cross\_account\_role\_external\_id) | AWP Cross account role external id | `string` | `null` (auto-generated) | no |
```
Comment (yuval-avram) [inline README.md:77]: same comment about "account settings"

File: main.tf [hunk 8]
```diff
@@ -1,15 +1,40 @@
...
+data "aws_region" "current" {
+  lifecycle {
+    postcondition {
+      condition     = self.name == local.region
+      error_message = "Error: AWP must be deployed in the same region as the CloudGuard Data Center: ${local.region} (Not in ${self.name})"
+    }
+  }
+}
+
+data "aws_organizations_organization" "org" {}
```
Comment (yuval-avram) [inline main.tf:19]: As discussed, can be removed?

File: README.md (hunk 6 above)
Comment (shayde78) [reply README.md:74]: done

File: variables.tf [hunk 9]
```diff
@@ -1,25 +1,30 @@
 variable "awp_cloud_account_id" {
-    description = "CLOUDGUARD_ACCOUNT_ID or EXTERNAL_AWS_ACCOUNT_NUMBER"
-    type        = string
+  description = "CLOUDGUARD_ACCOUNT_ID or EXTERNAL_AWS_ACCOUNT_NUMBER"
+  type        = string
+}
+
+variable "awp_centralized_cloud_account_id" {
+  description = "centralized account EXTERNAL_AWS_ACCOUNT_NUMBER or EXTERNAL_AWS_ACCOUNT_NUMBER"
```
Comment (batyakopelowitz) [inline variables.tf:7]: EXTERNAL_AWS_ACCOUNT_NUMBER or EXTERNAL_AWS_ACCOUNT_NUMBER ???
```suggestion
  description = "centralized account CLOUDGUARD_ACCOUNT_ID or EXTERNAL_AWS_ACCOUNT_NUMBER"
```

File: variables.tf [hunk 10]
```diff
@@ -1,25 +1,30 @@
...
+  type        = string
+  default     = null
 }
 
 variable "awp_scan_mode" {
-    description = "AWP scan mode <inAccount|saas>" # the valid values are "inAccount" and "saas" when onboarding the AWS account to Dome9 AWP.
-    type        = string
-    default     = "inAccount"
-    
+  description = "AWP scan mode <inAccount|saas|inAccountHub|inAccountSub>" # the valid values are "inAccount" and "saas" when onboarding the AWS account to Dome9 AWP.
```
Comment (batyakopelowitz) [inline variables.tf:13]: hub sub are also valid
```suggestion
  description = "AWP scan mode <inAccount|saas|inAccountHub|inAccountSub>" # the valid values are "inAccount", "saas" "inAccountHub" and "inAccountSub" when onboarding the AWS account to Dome9 AWP.
```

File: main.tf [hunk 11]
```diff
@@ -22,69 +47,62 @@ locals {
...
-
-# The `resource "aws_iam_role" "CloudGuardAWPCrossAccountRole"` block defines an IAM role that is used to allow CloudGuard AWP to access the AWS account.
-resource "aws_iam_role" "CloudGuardAWPCrossAccountRole" {
-  name        = var.awp_cross_account_role_name
-  description = "CloudGuard AWP Cross Account Role"
+# This policy provides the cross-account-role with the ability to read AWP scanner resources
+resource "aws_iam_policy" "CloudGuardAWPScannersReaderPolicy" {
+  count = local.is_scanner_mode_condition ? 1 : 0
+  name  = "CloudGuardAWPScannersReaderPolicy"
+  tags  = local.common_tags # TODO need this?, not appears in YAML?
```
Comment (batyakopelowitz) [inline main.tf:75]: remove comment

File: variables.tf (hunk 9 above)
Comment (shayde78) [reply variables.tf]: done thanks

File: variables.tf (hunk 10 above)
Comment (shayde78) [reply variables.tf:13]: updated thanks

File: main.tf (hunk 8 above)
Comment (shayde78) [reply main.tf:19]: the variable is removed,
the data is used by the module and not required the user for any input

File: README.md (hunk 4 above)
Comment (shayde78) [reply README.md:29]: done

File: README.md (hunk 7 above)
Comment (shayde78) [reply README.md:76]: done

**Review by batyakopelowitz — COMMENTED** [review]
update the provider version here: examples/complete/versions.tf
update the readme file
update the pr title

**Review by batyakopelowitz — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (shayde78) [reply README.md:35]: done

File: variables.tf (hunk 3 above)
Comment (shayde78) [reply variables.tf]: done

File: variables.tf [hunk 12]
```diff
@@ -1,25 +1,36 @@
...
+  type        = string
+  default     = ""
 }
 
 variable "awp_scan_mode" {
-    description = "AWP scan mode <inAccount|saas>" # the valid values are "inAccount" and "saas" when onboarding the AWS account to Dome9 AWP.
-    type        = string
-    default     = "inAccount"
-    
+  description = "AWP scan mode <inAccount|saas>" # the valid values are "inAccount" and "saas" when onboarding the AWS account to Dome9 AWP.
```
[the remark this answers is not in the document]
Comment (shayde78) [reply variables.tf]: done

File: variables.tf (hunk 2 above)
Comment (shayde78) [reply variables.tf]: done

File: variables.tf (hunk 1 above)
Comment (shayde78) [reply variables.tf]: renamed

**Review by yuval-avram — APPROVED** [review]

**Pull request merged**